packages/firestore/src/core/filter.ts

```diff
@@ -325,6 +325,14 @@ export function canonifyFilter(filter: Filter): string {
       filter.op.toString() +
       canonicalId(filter.value)
     );
+  } else if (compositeFilterIsFlatConjunction(filter)) {
+    // Older SDK versions use an implicit AND operation between their filters.
+    // In the new SDK versions, the developer may use an explicit AND filter.
+    // To stay consistent with the old usages, we add a special case to ensure
+    // the canonical ID for these two are the same. For example:
+    // `col.whereEquals("a", 1).whereEquals("b", 2)` should have the same
+    // canonical ID as `col.where(and(equals("a",1), equals("b",2)))`.
+    return filter.filters.map(filter => canonifyFilter(filter)).join(',');
```

Comment:
it's interesting that the canonical id of a=1 && b=1 is different strings in web and android.
In Android it'll be `a=1b=1`. In the Web, it'll be `a=1,b=1`. Nothing wrong with either one. I just checked `target.ts`  (canonifyTarget) to make sure this is correct.

Reply:
I had the same surprise. Tests caught it. Thanks for the thorough review!